arduino_mkrzero/src/lib.rs

```diff
@@ -0,0 +1,130 @@
+#![no_std]
+
+extern crate atsamd21_hal as hal;
+
+#[cfg(feature = "rt")]
+extern crate cortex_m_rt;
+#[cfg(feature = "rt")]
+pub use cortex_m_rt::entry;
+
+#[cfg(feature = "panic_abort")]
+pub extern crate panic_abort;
+
+pub use hal::atsamd21g18a::*;
+use hal::prelude::*;
+pub use hal::*;
+
+use gpio::{Floating, Input, Port};
+
+// This could possibly be simplified with concat_ident p for the name and P for the type
+// but I couldn't get it to work. Meta attribues (i.e. doc comments) are optional.
+macro_rules! input_pins {
+    // Insert trailing comma if it's missing
+    ($( $(#[$attr:meta])* pin $name:ident = $pin_name:ident: $pin_type:ident),+ ) =>
+        (input_pins!($( $(#[$attr])* pin $name = $pin_name: $pin_type),+ , ););
+
+    ($( $(#[$attr:meta])* pin $name:ident = $pin_name:ident: $pin_type:ident),+ , ) => {
+        /// Maps the pins to their arduino names and the numbers printed on the board.
+        /// Information from: https://github.com/arduino/ArduinoCore-samd/blob/master/variants/mkrzero/variant.cpp
+        pub struct Pins {
+            /// Opaque port reference
+            pub port: Port,
+            $($(#[$attr])* pub $name: gpio::$pin_type<Input<Floating>>),+
+        }
+
+        pub fn pins(port: atsamd21g18a::PORT) -> Pins {
```

Comment:
if/when this macro moves to the `hal` crate, this should probably become something like `port: hal::target_device::PORT`.